pkg/streamer/reader.go

```diff
@@ -202,6 +202,14 @@ func (r *BinlogReader) parseDirAsPossible(ctx context.Context, s *LocalStreamer,
 				if !strings.HasSuffix(relayLogFile, pos.Name) {
 					return false, "", "", terror.ErrFirstRelayLogNotMatchPos.Generate(relayLogFile, pos)
 				}
+				relayFilepath := path.Join(dir, relayLogFile)
+				cmp, err := fileSizeUpdated(relayFilepath, offset)
+				if err != nil {
+					return false, "", "", err
+				}
+				if cmp < 0 {
+					return false, "", "", terror.ErrRelayLogGivenPosTooBig
```

Comment:
there are two `for` loops in this function, so this error may not be caused by the meta file.

Reply:
Add a new function `checkRelayPos` to do that before loop.